lib/Cake/Network/Email/CakeEmail.php

```diff
@@ -1717,6 +1721,27 @@ protected function _renderTemplates($content) {
 			$rendered[$type] = implode("\n", $rendered[$type]);
 			$rendered[$type] = rtrim($rendered[$type], "\n");
 		}
+
+		/* Embed images inline in html templates */
+		if (!empty($rendered['html'])) {
```

Comment:
The memory concerns I had were more around serializing all the view variables with print_r() to do regexp on them. That `print_r` has the potential to allocate a significant amount of memory in creating a potentially very large string.

Memory concerns aside I don't think this kind of magic replacement behavior is something that belongs in a lower level class like `Mailer\Email`. I share @lorenzo's concern that it creates the opportunity for a security issue that I would rather not have. As a framework we can't be sure on the origin of email templates, and while the default rendering only reads templates off the filesystem the same can't be said for userland extensions/subclasses which may be sourcing their templates from user input.